api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -71,7 +80,7 @@ public boolean equals(Object other) {
     if (other == null || getClass() != other.getClass()) {
       return false;
     }
-
+    // not considering field id, as field-id will be reused.
```

Comment:
ID will be reused, but assignment is consistent because we assume that partition specs are not modified before the addition of partition field IDs. That means that tables start with only one spec that might not have IDs. Because we assign incrementally, IDs will always match when assigned using the default (1000, 1001, etc.).

Because we do have consistent IDs, I think this should check field ID here.